node/libs/crypto/src/bn254/hash.rs

```diff
@@ -1,27 +1,48 @@
 //! Hash operations.
 
+use ff_ce::{Field, PrimeField, SqrtField};
+use num_bigint::BigUint;
+use num_traits::Num;
 use pairing::{
-    bn256::{G1Affine, G1Compressed},
-    EncodedPoint,
+    bn256::{fq, Fq, FqRepr, Fr, G1Affine},
+    CurveAffine,
 };
 use sha3::Digest as _;
 
 /// Hashes an arbitrary message and maps it to an elliptic curve point in G1.
-pub(crate) fn hash_to_g1(msg: &[u8]) -> G1Affine {
-    for i in 0..256 {
-        // Hash the message with the index as suffix.
-        let bytes: [u8; 32] = sha3::Keccak256::new()
-            .chain_update(msg)
-            .chain_update((i as u32).to_be_bytes())
-            .finalize()
-            .into();
+pub(crate) fn hash_to_point(msg: &[u8]) -> (G1Affine, u8) {
+    let hash: [u8; 32] = sha3::Keccak256::new().chain_update(msg).finalize().into();
```

Comment:
64 bytes by IETF suggestion, or please cite the literature why non-uniform field element for X doesn't affect a protocol

Reply:
Why do we need 64 bytes? Maybe I'm misunderstanding something, but the prime field modulus is just 254 bits long. Generating a 64 bytes hash jus seems pointless.